### ISSUE TITLE
Clone of null dacl failes

### ISSUE THREAD

**Issue by liba2k** [kotauskas/interprocess#57]
When creating a named pipe as a service so any user process can connect. It is common to set a null dacl.
In this situation, the clone function in `try_clone.rs` fails.
This code triggers the bug:
```
use interprocess::local_socket::{ListenerOptions, ToNsName};
use interprocess::os::windows::{
    local_socket::ListenerOptionsExt, AsSecurityDescriptorMutExt, SecurityDescriptor,
};
fn main() {
    let mut sd = SecurityDescriptor::new().unwrap();
    unsafe {
        sd.set_dacl(std::ptr::null_mut(), false).unwrap();
    }
    let mut _conn = ListenerOptions::new()
        .name("test1".to_ns_name().unwrap())
        .security_descriptor(sd)
        .create_sync()
        .unwrap();
}
```
A possible solution looks can be: 
```
	let acl_fn = |(acl, dfl)| {
		io::Result::<(io::Result<LocalBox<ACL>>, bool)>::Ok((unsafe { clone_acl(acl) }, dfl))
	};
	let sid_fn = |(sid, dfl)| {
		io::Result::<(Option<LocalBox<c_void>>, bool)>::Ok((unsafe { clone_sid(sid)? }, dfl))
	};
	let dacl = old_sd.dacl()?.map(acl_fn).transpose()?;
	let sacl = old_sd.sacl()?.map(acl_fn).transpose()?;
	let owner = sid_fn(old_sd.owner()?)?;
	let group = sid_fn(old_sd.group()?)?;

	if let Some((acl, dfl)) = dacl {
		unsafe {
			if let Ok(acl) = acl {
				let mut acl = ManuallyDrop::new(acl);
				new_sd.set_dacl(acl.as_mut_ptr(), dfl)?
			} else {
				new_sd.set_dacl(ptr::null_mut(), dfl)?;
			}
		};
	}
	if let Some((acl, dfl)) = sacl {
		unsafe {
			if let Ok(acl) = acl {
				let mut acl = ManuallyDrop::new(acl);
				new_sd.set_dacl(acl.as_mut_ptr(), dfl)?
			} else {
				new_sd.set_dacl(ptr::null_mut(), dfl)?;
			}
		};
	}
```

Thank you for looking into it!

**Comment by kotauskas** [comment]
Fixed in [ffbf7c6](https://github.com/kotauskas/interprocess/commit/ffbf7c65911c87251c47a8403da9b91118eb259c).

**Comment by liba2k** [comment]
Thank you!!!
